sdk/core/core-client-rest/src/common.ts

```diff
@@ -94,6 +94,13 @@ export type RequestParameters = {
  */
 export type RawResponseCallback = (rawResponse: FullOperationResponse, error?: unknown) => void;
 
+/**
+ * Types of bodies supported on the response.
+ * NodeJS.ReadableStream and () =\> NodeJS.ReadableStream is Node only.
+ * Blob, ReadableStream<Uint8Array>, and () =\> ReadableStream<Uint8Array> are browser only.
+ */
+export type ResponseBodyType = RequestBodyType | Record<string, unknown>;
```

Comment:
Some types may not be applicable (e.g., I don't we would return a body of `Blob` or `FormData`?)  Although changing it looks breaking...

Reply:
Oh you're right, seems like `parsedBody` is meant to contain the parsed JSON representation of the body only. I am not sure why do we put the string representation there if parsing fails. The string representation is easily accessible through `bodyAsText` property on the same object.

Reply:
Given that the only way to consume this property properly is through casting, how do we feel about applying the breaking change anyway? I think the most accurate type would be `Record<string, unknown> | string`. /cc @xirzec

Reply:
We can even go further and ditch the `string` which I think is the true intent of this property but that would be truly breaking if someone uses this property for a `text/plain` response.

Reply:
I went with `Record<string, unknown> | string` in https://github.com/Azure/azure-sdk-for-js/pull/27000/commits/3e49a31c2abcb74c8135c5e5f1b68572a8a132bd.